subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/graph/DependencyGraphBuilder.java

```diff
@@ -240,6 +241,20 @@ public void execute(BuildOperationQueue<RunnableBuildOperation> buildOperationQu
         });
     }
 
+    private void resolveModuleRevisionIdsConcurrently(final List<EdgeState> dependencies) {
+        if (dependencies.size() <= 1) {
+            return;
+        }
+        buildOperationExecutor.runAll(new Action<BuildOperationQueue<RunnableBuildOperation>>() {
+            @Override
+            public void execute(BuildOperationQueue<RunnableBuildOperation> buildOperationQueue) {
+                for (final EdgeState dependency : dependencies) {
+                    buildOperationQueue.add(new ResolveComponentIdOperation(dependency.selector));
```

Comment:
We should make sure that we don't run these in parallel if they're already resolved. Parallel noops are slower than serial.